__init__.py

```diff
@@ -662,6 +716,143 @@ def _get_alarm_matches(
         # No matches found
         return (status[2], None)
 
+    # sometimes LF returns a roman numeral for
+    # a number so the 26th will return xxvi
+    def roman_to_int(self, s):
+        rom_val = {'I': 1, 'V': 5, 'X': 10, 'L': 50, 'C': 100, 'D': 500, 'M': 1000}
+        int_val = 0
+        for i in range(len(s)):
+            if i > 0 and rom_val[s[i]] > rom_val[s[i - 1]]:
+                int_val += rom_val[s[i]] - 2 * rom_val[s[i - 1]]
+            else:
+                int_val += rom_val[s[i]]
+        return int_val
+
+    def get_advanced_matches(self, utt, have_time, when, when_utc, user_dow, alarm_list):
```

Comment:
this is a really long method.  could it be broken up?

Reply:
Not easily, however I notice it is about 60 lines long while the weather skill has a method called  handle_week_weather() which is about 90 lines long; whats the diff?

Reply:
not in my refactor branch 😄